src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -91,10 +105,18 @@ public Column(string name, DataKind? type, Range[] source, KeyCount keyCount = n
             public string Name;
 
             /// <summary>
-            /// <see cref="DataKind"/> of the items in the column. If <see langword="null"/> defaults to a float.
+            /// <see cref="InternalDataKind"/> of the items in the column. It defaults to float.
+            /// Although <see cref="InternalDataKind"/> is internal, <see cref="Type"/>'s information can be publically accessed by <see cref="DataKind"/>.
             /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Type of the items in the column")]
-            public DataKind? Type;
+            [BestFriend]
+            internal InternalDataKind Type = InternalDataKind.R4;
+
+            /// <summary>
+            /// <see cref="Data.DataKind"/> of the items in the column.
+            /// </summary>
+            /// It's a public interface to access the information in an internal DataKind.
+            public DataKind DataKind => Type.ToDataKind();
```

Comment:
I think this should be settable. All the other fields are settable on this class:

`Name`, `KeyCount`, `Source`, but this one isn't. So if I use the default constructor - there is no way to set this property.

Reply:
Ok. Will make it
```csharp
            public DataKind DataKind
            {
                get { return Type.ToDataKind(); }
                set { Type = value.ToInternalDataKind(); }
            }
```

---
In reply to: [259954397](https://github.com/dotnet/machinelearning/pull/2661#discussion_r259954397) [](ancestors = 259954397)